libs/openchallenges/challenge-search/src/lib/challenge-search.component.ts

```diff
@@ -169,6 +169,12 @@ export class ChallengeSearchComponent
     //       active: false,
     //     })))
     // );
+
+    // update the total number of challenges in database
+    this.challengeService
+      .listChallenges(this.query as ChallengeSearchQuery)
+      .subscribe((page) => (this.totalChallengesCount = page.totalElements));
+
```

Comment:
@tschaffter I assume here if `pageSize` is `undefined`, then all results will be returned - but let me know if it's not correct.

Reply:
I think we should set a default of 20 maybe?

Reply:
To clarify, this purpose of snippet is to retrieve the total number of challenges that we have in the database and will be displayed in this element:
 
<img width="189" alt="Screen Shot 2023-01-24 at 3 11 33 PM" src="https://user-images.githubusercontent.com/73901500/214398219-159a8628-d771-422f-8982-f4107dfc9f23.png">

>  I think we should set a default of 20 maybe?

There is also a `defaultQuery` where we currently use `50` as default `pageSize`, but it reminds me that the pagination of this page has been added. Do we limit how many challenge cards per line for this page?

Reply:
Mine is scaled to that too :-)

Reply:
The size of the page is defined by the property [`size`](https://github.com/tschaffter/sage-monorepo/blob/add-kafka-admin/libs/openchallenges/api-description/src/components/schemas/PageMetadata.yaml). The value of this property cannot be undefined.

Reply:
> the purpose of this snippet is to retrieve the total number of challenges that we have in the database and will be displayed in below element.

The total number of challenges can be obtained by sending an empty query object and then extract the value of the page property `totalElements`.

Reply:
> Can you remind me of the size of your big screen (mine is 3840 × 2160)?

Mine too but the design should not rely on user having 4K monitor. :-)

Reply:
> The total number of challenges in the DB can be obtained by sending an empty query object

ah, that's easier.

> Can you remind me of the size of your big screen (mine is 3840 × 2160)?
Mine too but the design should not rely on user having 4K monitors.

Cool! yeah, I agree the design will reply on breakpoints, instead of 4k - that's too large, but at least we know we are now viewing at the same size~